book/src/http_beacon.md

```diff
@@ -183,9 +205,212 @@ Typical Responses | 200, 404
+    "previous_justified_slot": 37856,
+    "previous_justified_block_root": "0xbdae152b62acef1e5c332697567d2b89e358628790b8273729096da670b23e86"
+}
+```
+
+## `/beacon/heads`
+
+Returns the roots of all known head blocks. Only one of these roots is the
+canonical head and that is decided by the fork choice algorithm. See [`/beacon/head`](#beaconhead) for the canonical head.
+
+### HTTP Specification
+
+| Property | Specification |
+| --- |--- |
+Path | `/beacon/heads`
+Method | GET
+JSON Encoding | Object
+Query Parameters | None
+Typical Responses | 200
+
+### Example Response
+
+```json
+[
+    {
+        "beacon_block_root": "0x226b2fd7c5f3d31dbb21444b96dfafe715f0017cd16545ecc4ffa87229496a69",
+        "beacon_block_slot": 38373
+    },
+    {
+        "beacon_block_root": "0x41ed5b253c4fc841cba8a6d44acbe101866bc674c3cfa3c4e9f7388f465aa15b",
+        "beacon_block_slot": 38375
+    }
+]
+```
+
+## `/beacon/proposer_slashing`
+
+Adds the specified `proposer_slashing` to the operation pool.
+
+### HTTP Specification
+
+| Property | Specification |
+| --- |--- |
+Path | `/beacon/proposer_slashing`
+Method | POST
+JSON Encoding | Object
+Query Parameters | None
+Typical Responses | 200
+
+### Request Body
+
+Expects the following object in the POST request body:
+
+```
+{
+    proposer_index: u64,
+    header_1: {
+        slot: Slot,
+        parent_root: Bytes32,
+        state_root: Bytes32,
+        body_root: Bytes32,
+        signature: Bytes32
+    },
+    header_2: {
+        slot: Slot,
+        parent_root: Bytes32,
+        state_root: Bytes32,
+        body_root: Bytes32,
+        signature: Bytes32
+    }
+}
+```
+
+### Returns
+
+Returns `true` if the proposer slashing was inserted successfully, or the corresponding error if it failed.
+
+### Example
+
+### Request Body
+
+```json
+{
+	"proposer_index": 0,
+    "header_1": {
+        "slot": 0,
+        "parent_root": "0x0101010101010101010101010101010101010101010101010101010101010101",
+        "state_root": "0x0101010101010101010101010101010101010101010101010101010101010101",
+        "body_root": "0x0101010101010101010101010101010101010101010101010101010101010101",
+        "signature": "0xb8970d1342c6d5779c700ec366efd0ca819937ca330960db3ca5a55eb370a3edd83f4cbb2f74d06e82f934fcbd4bb80609a19c2254cc8b3532a4efff9e80edf312ac735757c059d77126851e377f875593e64ba50d1dffe69a809a409202dd12"
+    },
+    "header_2": {
+        "slot": 0,
+        "parent_root": "0x0202020202020202020202020202020202020202020202020202020202020202",
+        "state_root": "0x0101010101010101010101010101010101010101010101010101010101010101",
+        "body_root": "0x0101010101010101010101010101010101010101010101010101010101010101",
+        "signature": "0xb60e6b348698a34e59b22e0af96f8809f977f00f95d52375383ade8d22e9102270a66c6d52b0434214897e11ca4896871510c01b3fd74d62108a855658d5705fcfc4ced5136264a1c6496f05918576926aa191b1ad311b7e27f5aa2167aba294"
+    }
+}
```

Comment:
The proposer_slashing inserted here will be valid (return `true`) but will make the beacon client behave strangely due to some invalid data being inserted (stopping block production since block header won't be valid anymore...). Should we find a way to explain it?